### PR TITLE
[PATCH-1933] Fixed Data Layer handling of SAP parameter list field de…

### PR DESCRIPTION
**Pull request by punkhorn**
…faults.

### CODE REVIEW

File: camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java [hunk 1]
```diff
@@ -1345,4 +1345,220 @@ public static String checkForKeywordLiterals(String value) {
...
+			return convertJCoBinaryDefaultValue(listMetaData.getDefault(index));
+
+		case JCoMetaData.TYPE_STRING:
+			return convertJCoStringDefaultValue(listMetaData.getDefault(index));
+
+		case JCoMetaData.TYPE_XSTRING:
+			return convertJCoBinaryDefaultValue(listMetaData.getDefault(index));
+
+		case JCoMetaData.TYPE_DECF16:
+			return convertJcoDecimalDefaultValue(listMetaData.getDefault(index));
```
Comment (apupier) [inline camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java]: What do you think to merge with JCoMetaData.TYPE_DECF34 ?

File: camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java [hunk 2]
```diff
@@ -1345,4 +1345,220 @@ public static String checkForKeywordLiterals(String value) {
...
+		case JCoMetaData.TYPE_FLOAT:
+			return convertJCoFloatDefaultValue(listMetaData.getDefault(index));
+
+		case JCoMetaData.TYPE_BYTE:
+			return convertJCoBinaryDefaultValue(listMetaData.getDefault(index));
+
+		case JCoMetaData.TYPE_STRING:
+			return convertJCoStringDefaultValue(listMetaData.getDefault(index));
+
+		case JCoMetaData.TYPE_XSTRING:
```
Comment (apupier) [inline camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java]: What do you think to merge with JCoMetaData.TYPE_BYTE ?

File: camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java [hunk 3]
```diff
@@ -1345,4 +1345,220 @@ public static String checkForKeywordLiterals(String value) {
...
+		case JCoMetaData.TYPE_INT:
+		case JCoMetaData.TYPE_INT1:
+		case JCoMetaData.TYPE_INT2:
+			return convertJcoIntegerDefaultValue(listMetaData.getDefault(index));
+
+		case JCoMetaData.TYPE_CHAR:
+		case JCoMetaData.TYPE_NUM:
+			return convertJCoStringDefaultValue(listMetaData.getDefault(index));
+
+		case JCoMetaData.TYPE_BCD:
```
Comment (apupier) [inline camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java]: What do you think to merge with JCoMetaData.TYPE_DECF34 and JCoMetaData.TYPE_DECF16 ?

File: camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java [hunk 4]
```diff
@@ -1345,4 +1345,220 @@ public static String checkForKeywordLiterals(String value) {
...
+	private static byte[] convertJCoBinaryDefaultValue(String jcoDefaultBinaryValue) {
+		if (jcoDefaultBinaryValue == null) {
+			return null;
+		}
+		jcoDefaultBinaryValue = stripSingleQuotePrefixSuffix(jcoDefaultBinaryValue);
+		try {
+			int length = jcoDefaultBinaryValue.length();
+			byte[] binaryData = new byte[length / 2];
+			for (int i = 0; i< length; i += 2) {
+				binaryData[i / 2] = (byte) ((Character.digit(jcoDefaultBinaryValue.charAt(i), 16) << 4) +
```
Comment (apupier) [inline camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java]: would be nice to write several unit test for this use case because personally i'm unable to understand what it is doing (and then to understand if corner cases are handled)

**Comment by paoloantinori** [comment]
in `camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java`
```
BigDecimal cannot be resolved to a type
Date cannot be resolved to a type
```

*Toto, we're not using Groovy anymore.*

**Comment by punkhorn** [comment]
@paoloantinori git cherry-pick victim on 'BigDecimal cannot be resolved to a type
Date cannot be resolved to a type'. Updated and thanks for catching!

File: camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java (hunk 4 above)
Comment (punkhorn) [inline camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java]: @apupier this is converting the JCo string representation of a byte array (a Hex String prefixed and suffixed with a single quote) into the equivalent byte array. Good idea to add a unit test for this. Thanks!

**Comment by punkhorn** [comment]
@apupier yes these should be consolidated. I will do that in next check-in on master. Thanks!
